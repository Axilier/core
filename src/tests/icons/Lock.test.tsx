import React from 'react';
import { mount } from 'enzyme';
import { Lock } from '../../index';

const onChange = jest.fn();
const lock = mount(<Lock className={'test-Lock'} />);

describe('when I render the component in an application', () => {
    it('should render correctly', () => {
        expect(lock).toMatchSnapshot();
    });
    describe('when the `locked` prop is not defined', () => {
        it('should default to the locked state', () => {
            expect(
                lock.find('.test-Lock').hostNodes().childAt(1).props().name,
            ).toBe('locked');
        });
    });
    describe('when I change the `locked` prop it should update the component', () => {
        beforeAll(() => {
            lock.setProps({ locked: false });
            lock.update();
        });
        it('should change the components name to unlocked', () => {
            expect(
                lock.find('.test-Lock').hostNodes().childAt(1).props().name,
            ).toBe('unlocked');
        });
    });
    describe('when the lock is `disabled`', () => {
        beforeAll(() => {
            lock.setProps({ disabled: true });
        });
        it('should render a disabled component', () => {
            expect(lock.find('.disable-slash').exists()).toBeTruthy();
        });
    });
});
