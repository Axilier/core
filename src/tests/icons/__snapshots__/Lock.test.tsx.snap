// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when I render the component in an application should render correctly 1`] = `
<Lock
  className="test-Lock"
  locked={true}
>
  <svg
    className="test-Lock icon lock"
    clipRule="evenodd"
    fillRule="evenodd"
    strokeLinejoin="round"
    strokeMiterlimit={2}
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h16v16H0z"
      fill="none"
    />
    <path
      d="M4.587 14.066a1.139 1.139 0 01-1.144-1.144V7.585c0-.636.508-1.144 1.144-1.144h.38V4.916c0-1.652 1.399-3.05 3.05-3.05 1.653 0 3.05 1.398 3.05 3.05v1.525h.382c.636 0 1.144.508 1.144 1.144v5.337c0 .636-.508 1.144-1.144 1.144H4.587zm0-1.144h6.862V7.585H4.587v5.337zm5.337-6.481V4.916c0-1.017-.89-1.906-1.906-1.906-1.017 0-1.906.89-1.906 1.906v1.525h3.812z"
      fill="black"
      fillRule="nonzero"
      name="locked"
    />
  </svg>
</Lock>
`;
